src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java

```diff
@@ -516,4 +923,89 @@ public String validateGiverRecipientVisibility(FeedbackQuestionAttributes feedba
 
         return answerFrequency;
     }
+
+    /**
+     * Calculates the average weight of each option based on the all responses.
+     * Weights should be assigned or else an empty map will be returned.<br>
+     * The average of an option is calculated as follows:<br><br>
+     * Average = (frequency of the option based on responses) * (weight of the option) / total responses.
```

Comment:
@xpdavid To be honest, I am not sure what is considered the correct way of using the weight. But this portion of code (Calculating the average for each option based on their weight and response count) is similar to what we show in rubric question stats. 
![screenshot 528](https://user-images.githubusercontent.com/10681755/41614515-53883b7e-7416-11e8-9c8d-bb78ecec402a.png)

Reply:
@damithc Is this the correct way of calculating weights?

Reply:
@sukanta-27 can give a specific example scenario (question, options, responses) and the stats calculated? That way we can get a sense of whether the stats make sense.

Reply:
@damithc @xpdavid Here is an example for MCQ question

**Question:**
![screenshot 529](https://user-images.githubusercontent.com/10681755/41645564-e4f7d688-7460-11e8-882e-5667d20321b0.png)

**Responses:**
![screenshot 531](https://user-images.githubusercontent.com/10681755/41645628-0929f036-7461-11e8-8b75-3c7a93fb41ae.png)

**Stats:**
![screenshot 530](https://user-images.githubusercontent.com/10681755/41645669-277b0aca-7461-11e8-9449-7f1b0362a4ab.png)

Reply:
@sukanta-27 thanks. the doubt is regarding the `Response Summary` section and not the `Per Recipient Statistics`? Which columns?

Reply:
>the doubt is regarding the Response Summary section and not the Per Recipient Statistics?

@damithc Yes, I guess the doubt is regarding the `Response Summary` section, Average coulmn. @xpdavid Can you confirm?

Reply:
Yes. Actually, I am curious why the denominator doesn't have the `totalWeight` factor.

Reply:
@xpdavid You mean to say:
`Average = (response count per option) * (weight of that option) / total Weight`?

Reply:
It's hard to give a meaning to the `Average` column. I suspect users will be equally confused about what it means. Perhaps omit `Weight` and `Average` columns altogether and show only `Percentage(count)`? i.e., weights are not considered in this table at all.

Reply:
Something like that. I think as long as the stats help instructors to make decisions, it should be good. 

Actually, weighted Percentage can also be a good option.

WeightedPercentage = (response count per option) * (weight of that option) / [(response count for option i) * (weight of option i) for i to n]

Let's say if no weight (same as weight = 1), this will be the same as a percentage. If some option is not helpful (like `Not Sure` in feedback), instructors can give it lower weight. We can also see the difference between normal percentage and weighted percentage.

Reply:
I think having two kind of percentage can confuse the instructor? Isn't that so? Specially when we don't show that kind of stat anywhere else.

Reply:
>Actually, weighted Percentage can also be a good option.

Yes, that can be useful.

Reply:
> I think having two kind of percentage can confuse the instructor?

We'll have to clearly indicate which column is weighted and which is not. Actually, it's best to show both or none at all; otherwise the user will wonder which one it is, misinterpret one as the other, or not even realize there are two types of percentages possible.

Reply:
@xpdavid @damithc Okay, in that case, I will replace the `Average` column with `Weighted percentage` column. Will also add some details about the weighted percentage in the instructor help page.